homeassistant/components/xiaomi_miio/number.py

```diff
@@ -0,0 +1,214 @@
+"""Support for Xiaomi Mi Air Purifier and Xiaomi Mi Air Humidifier."""
+from dataclasses import dataclass
+from enum import Enum
+import logging
+
+from homeassistant.components.number import NumberEntity
+from homeassistant.const import CONF_HOST, CONF_TOKEN
+from homeassistant.core import callback
+
+from .const import (
+    CONF_DEVICE,
+    CONF_FLOW_TYPE,
+    CONF_MODEL,
+    DOMAIN,
+    FEATURE_FLAGS_AIRHUMIDIFIER,
+    FEATURE_FLAGS_AIRHUMIDIFIER_CA4,
+    FEATURE_FLAGS_AIRHUMIDIFIER_CA_AND_CB,
+    FEATURE_SET_MOTOR_SPEED,
+    KEY_COORDINATOR,
+    KEY_DEVICE,
+    MODEL_AIRHUMIDIFIER_CA1,
+    MODEL_AIRHUMIDIFIER_CA4,
+    MODEL_AIRHUMIDIFIER_CB1,
+    MODELS_HUMIDIFIER,
+    SERVICE_SET_MOTOR_SPEED,
+)
+from .device import XiaomiCoordinatedMiioEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_MOTOR_SPEED = "motor_speed"
+ATTR_ACTUAL_MOTOR_SPEED = "actual_speed"
+
+SERVICE_TO_METHOD = {
+    SERVICE_SET_MOTOR_SPEED: {
+        "method": "async_set_motor_speed",
+        "property": ATTR_MOTOR_SPEED,
+    },
+}
+
+
+@dataclass
+class NumberType:
+    """Class that holds device specific info for a xiaomi aqara or humidifier number controller types."""
+
+    name: str = None
+    short_name: str = None
+    unit_of_measurement: str = None
+    icon: str = None
+    device_class: str = None
+    min: float = None
+    max: float = None
+    step: float = None
+    service: str = None
+    available_with_device_off: bool = True
+
+
+NUMBER_TYPES = {
+    FEATURE_SET_MOTOR_SPEED: NumberType(
+        name="Motor Speed",
+        icon="mdi:fast-forward-outline",
+        short_name=ATTR_MOTOR_SPEED,
+        unit_of_measurement="rpm",
+        min=200,
+        max=2000,
+        step=10,
+        service=SERVICE_SET_MOTOR_SPEED,
+        available_with_device_off=False,
+    ),
+}
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Selectors from a config entry."""
+    entities = []
+    if config_entry.data[CONF_FLOW_TYPE] == CONF_DEVICE:
```

Comment:
Please invert the logic and return if true. Then we can outdent below.

Reply:
Done